### ISSUE TITLE
Zoom in/out of devices on trackpad pinch gesture in the preview area

### ISSUE THREAD

**Issue by manojVivek** [responsively-org/responsively-app#1049]

**Comment by prabhjotlamba1655** [comment]
Hi @manojVivek, I would like to work on this issue. Please assign it to me.

**Comment by manojVivek** [comment]
@prabhjotlamba1655 Please go ahead!

**Comment by dgbkn** [comment]
HI @manojVivek i would like to take up this issue if possible.

**Comment by manojVivek** [comment]
@dgbkn Please go for it! 
We stopped doing user assignments for the issues.

**Comment by Priyanshu72** [comment]
can you please elaborate this issue

**Comment by MarCUS103** [comment]
@manojVivek Could you please elaborate the issue? It will be very beneficial while resolving the issue.

**Comment by mukulpythondev** [comment]
I am start working on this issue.

**Comment by sameerbiradar2001** [comment]
Hi @manojVivek, I would like to work on this issue. Please assign it to me.

**Comment by manojVivek** [comment]
@Alok-Singh-ops Thanks for exploring that. 

But we don't actually need that event handler on the webview, we just have to add it to the preview area enclosing the webview.

**Comment by Alok-Singh-ops** [comment]
@manojVivek yes i get it. i Even tried to do using lib such as zoom pan but I guess the library just does not support webview

**Comment by manojVivek** [comment]
Hm, I'm not certain why webview would affect anything here. 

As we don't need to interact with the webview at all in this, we just have to watch for the events on the preview container and update the zoom level in the redux state.

**Comment by Alok-Singh-ops** [comment]
@manojVivek Gotcha. By this entire preview section will be zoomed in/zoomed out

**Comment by Alok-Singh-ops** [comment]
@manojVivek  Also, I have raised by PR kindly look into it

**Comment by manojVivek** [comment]
> By this entire preview section will be zoomed in/zoomed out

Just clarifying, not the entire section, just the devices will be zoomed by the `zoomFactor` state value.

**Comment by dompolochak** [comment]
From what I can tell there needs to be some sort of onWheel or onScroll event to trigger the zoomFactor update but it doesnt seem to respond to any of those events (also doesn't respond to onClick) for some reason I can't figure out
<img width="420" alt="Screenshot 2023-12-04 at 2 59 37 PM" src="https://github.com/responsively-org/responsively-app/assets/48055644/0c7e7af2-11c0-4a36-8dd8-90560aa13dd7">

**Comment by 123upasana** [comment]
Hi @manojVivek, I would like to work on this issue. Please assign it to me.

**Comment by lyncasterc** [comment]
Hey @manojVivek, I went ahead and created a PR for this (#1196) since it looks like it hasn't been addressed yet!

**Comment by LungsomLamnio** [comment]
> _No description provided._

can you please provide a description

**Comment by Shashikant2707** [comment]
Hi @manojVivek, I would like to work on this issue

**Comment by manojVivek** [comment]
Please go ahead.

**Comment by mayanknetam1234** [comment]
hi @manojVivek I would like to tackle this issue , please assign it to me

**Comment by tavcoder** [comment]
Hi! I'd love to work on this issue. Please assign it to me

**Comment by Prathameshw4** [comment]
Hi I would like to work on this issue. Could you please assign it to me?

**Comment by ashmeetsingh9119** [comment]
<div style="display: flex; flex-direction: row;align-items: center;margin: 0px 50px;justify-content: center;">
        <div style="width: 50%;" id="preview-area">
            <h1 style="color: rgb(0, 83, 150);margin-bottom: 10px;">Pinch Preview Zoom In Zoom Out</h1>
            <p style="color: rgb(77, 77, 77);margin-bottom: 10px;">Lorem ipsum dolor sit amet consectetur adipisicing elit. Non aperiam hic excepturi quisquam aspernatur quo consequatur quia, quas nihil velit impedit rerum laborum quam porro animi illo, ex enim mollitia saepe dolores eaque iure. Inventore itaque corrupti iste explicabo sint, doloribus blanditiis quae aliquam reprehenderit nemo, quia libero, necessitatibus adipisci!</p>
        </div>
        <div style="width: 50%;">
            <img src="https://images.unsplash.com/photo-1556742049-0cfed4f6a45d?ixlib=rb-4.0.3&auto=format&fit=crop&w=800&q=80" alt="Project Screenshot" class="img-fluid rounded shadow-lg" style="width: 400px;">
        </div>
    </div>
    
     <script>
        document.addEventListener('wheel', function(e) {
            if (e.ctrlKey) {
                e.preventDefault();
            }
        }, { passive: false });

        let scale = 1;
        const preview = document.getElementById('preview-area');

        preview.style.transformOrigin = 'center center'; // Or "top left" if preferred

        preview.addEventListener('wheel', function(e) {
            if (e.ctrlKey || e.metaKey) {
                e.preventDefault();

                const zoomSpeed = 0.05;
                if (e.deltaY < 0) {
                    // Zoom in
                    scale = Math.min(3, scale + zoomSpeed);
                } else {
                    // Zoom out
                    scale = Math.max(1, scale - zoomSpeed); // Prevent going below actual size
                }

                preview.style.transform = `scale(${scale})`;
            }
        }, { passive: false });
    </script>

<img width="1647" height="395" alt="Image" src="https://github.com/user-attachments/assets/06440d9d-344b-4821-9403-4548af99636d" />

**Comment by BboyGT** [comment]
I opened a focused PR for this here: #1498.

The change listens for pinch-style wheel events on each preview webview and maps them to the existing zoom in/out actions, with a small threshold so trackpads do not jump through zoom levels too aggressively. Happy to adjust the behavior if you prefer a different sensitivity.

**Comment by oruccoder** [comment]
Hi @manojVivek, I would like to work on this issue. Please assign it to me.

**Comment by anrey13** [comment]
Hi! I'd like to work on this issue. If it's still available, could you please assign it to me?

Thank you!
